pkg/test/framework/components/policybackend/policybackend.go

```diff
@@ -118,11 +121,19 @@ func (c *localComponent) Init(ctx environment.ComponentContext, deps map[depende
 		return nil, fmt.Errorf("unsupported environment: %q", ctx.Environment().EnvironmentID())
 	}
 
-	port := policy.DefaultPort // TODO: Allow dynamically allocated ports.
+	pm, err := reserveport.NewPortManager()
```

Comment:
Could you just have the policy backend dynamically assign the port and provide access to it after `Start`?